src/com/google/enterprise/gsafeed/Acl.java

```diff
@@ -70,24 +106,45 @@ public Acl setUrl(String value) {
   /**
    * Gets the value of the inheritanceType property.
    *
-   * @return possible object is {@link String}
+   * @return possible object is {@link InheritanceType}
    */
-  public String getInheritanceType() {
+  public InheritanceType getInheritanceType() {
     if (inheritanceType == null) {
-      return "leaf-node";
+      return null;
     } else {
-      return inheritanceType;
+      return InheritanceType.fromString(inheritanceType);
     }
   }
 
   /**
    * Sets the value of the inheritanceType property.
    *
-   * @param value allowed object is {@link String}
+   * @param value allowed object is {@link String} or null
    * @return this object
+   * @throws IllegalArgumentException if value isn't a valid
+   * inheritance-type attribute value
    */
   public Acl setInheritanceType(String value) {
-    this.inheritanceType = value;
+    if (Strings.isNullOrEmpty(value)) {
```

Comment:
What's the purpose of allowing empty strings to act like nulls?

Maybe, if we don't allow empty strings, and we keep the string representation of enums, we could move the null check into fromString and remove it from the getter and String setter.